datadog/resource_datadog_dashboard.go

```diff
@@ -4531,6 +4531,255 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
 	return terraformDefinition
 }
 
+func getFormulaSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"formula_expression": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "String expression built from queries, formulas and functions.",
+				},
+				"limit": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					Description: "Options for limiting results returned.",
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"count": {
+								Type:     schema.TypeInt,
+								Optional: true,
+							},
+							"order": {
+								Type:     schema.TypeString,
+								Optional: true,
+								Default:  "desc",
+							},
+						},
+					},
+				},
+				"alias": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Expression alias.",
+				},
+			},
+		},
+	}
+}
+
+func getFormulaQuerySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"metric_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "A timeseries formula and functions metrics query.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Data source for metrics queries.",
```

Comment:
This is a static value right? (Will always be `metric`? 
Do you think it'd make sense to remove this from being Required and just set the Default?
https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#default

Reply:
For a metric query, the value will always be `metrics`. I've made the related changes in commit [5c22cd7](https://github.com/DataDog/terraform-provider-datadog/pull/892/commits/5c22cd771533e25ac58a9e87bc9da024672d34f3).